libyul/backends/wasm/WordSizeTransform.cpp

```diff
@@ -100,9 +100,9 @@ void WordSizeTransform::operator()(Block& _block)
 
 					// Special handling for datasize and dataoffset - they will only need one variable.
 					if (BuiltinFunction const* f = m_inputDialect.builtin(std::get<FunctionCall>(*varDecl.value).functionName.name))
-						if (!f->literalArguments.empty() && f->literalArguments[0])
+						if (f->name == "datasize"_yulstring || f->name == "dataoffset"_yulstring)
```

Comment:
`datasize` and `dataoffset` are currently the only functions taking literal arguments and I guess this fact was used here instead of checking directly for the names.

I changed this because it most likely will no longer be true once we add the immutable builtins, so now we check for the names and assert that they are literals only.

Reply:
This was due to `datasize` and `dataoffset` just returning a single `i64` in the wasm dialect. So it is more about the return type than the parameters.

Reply:
But you are right, checking for the name is the better option.